Formula/rosa-cli.rb

```diff
@@ -0,0 +1,31 @@
+class RosaCli < Formula
+  desc "RedHat OpenShift Service on AWS (ROSA) command-line interface"
+  homepage "https://www.openshift.com/products/amazon-openshift"
+  url "https://github.com/openshift/rosa.git",
+    tag:      "v1.0.1",
+    revision: "585ab60484929ffdad1f4b83ea29acca6c6e88f7",
+    shallow:  false
+  license "Apache-2.0"
+
+  head "https://github.com/openshift/rosa.git", shallow: false
+
+  livecheck do
+    url :stable
+    regex(/^v(\d+\.\d+\.\d+)$/i)
+  end
+
+  depends_on "coreutils" => :build
+  depends_on "go" => :build
+  depends_on "awscli"
+
+  def install
+    system "make"
+    bin.install "rosa"
+    output = Utils.safe_popen_read("#{bin}/rosa", "completion")
+    (bash_completion/"rosa").write output
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/rosa version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.